File: promemoria.py
# Dichiarazione delle variabili
numero = 10
nome = "John"
salario = 1000.50

# Dichiarazione di una funzione
def stampa_messaggio():
    print("Ciao, mondo!")

# Condizionali if-else
eta = 25

if eta >= 18:
    print("Sei maggiorenne.")
else:
    print("Sei minorenne.")

# Ciclo for
for i in range(5):
    print(i)

# Ciclo while
contatore = 0
while contatore < 5:
    print(contatore)
    contatore += 1

# Switch - In Python, il costrutto switch non esiste nativamente, ma puoi simulare lo stesso comportamento con un dizionario.
print("Inserisci un giorno della settimana (1-7):")
giorno = int(input())
nomi_giorni = {
    1: "Lunedì",
    2: "Martedì",
    3: "Mercoledì",
    4: "Giovedì",
    5: "Venerdì",
    6: "Sabato",
    7: "Domenica"
}
nome_giorno = nomi_giorni.get(giorno, "Giorno non valido")
print(f"Il giorno corrispondente al numero {giorno} è {nome_giorno}.")

# Dichiarazione di una lista
nomi = ["Alice", "Bob", "Charlie"]

# Accesso agli elementi di una lista
primo_nome = nomi[0]
secondo_nome = nomi[1]

# Modifica degli elementi di una lista
nomi[2] = "Eve"

# Ottenere la lunghezza di una lista
lunghezza_nomi = len(nomi)

# Iterazione attraverso una lista
for nome in nomi:
    print(nome)

# Array multidimensionali - In Python, è possibile utilizzare liste annidate per creare una struttura simile a una matrice
matrice = [[0, 0, 0], [0, 0, 0], [0, 0, 0]]
matrice[0][0] = 1
matrice[1][1] = 2
valore = matrice[2][2]

# Classe 
class Persona:
    def __init__(self, nome, eta):
        self.nome = nome
        self.eta = eta
#oggetto
persona1 = Persona("Maria", 30)

File: promemoria.php
<?php
// Dichiarazione di variabili
$numero = 10;
$nome = "John";
$salario = 1000.50;

// Dichiarazione di una funzione
function stampareMessaggio() {
    echo "Ciao, mondo!";
}

// Condizionali if-else
$eta = 25;

if ($eta >= 18) {
    echo "Sei maggiorenne.";
} else {
    echo "Sei minorenne.";
}

// Ciclo for
for ($i = 0; $i < 5; $i++) {
    echo $i;
}

// Ciclo while
$contatore = 0;
while ($contatore < 5) {
    echo $contatore;
    $contatore++;
}

// Switch
echo "Inserisci un giorno della settimana (1-7):";
$giorno = intval(readline());
$nomeGiorno = "";

switch ($giorno) {
    case 1:
        $nomeGiorno = "Lunedì";
        break;
    
    case 2:
        $nomeGiorno = "Martedì";
        break;
            
    case 3:
        $nomeGiorno = "Mercoledì";
        break;
    
    case 4:
        $nomeGiorno = "Giovedì";
        break;
    
    case 5:
        $nomeGiorno = "Venerdì";
        break;
    
    case 6:
        $nomeGiorno = "Sabato";
        break;
    
    case 7:
        $nomeGiorno = "Domenica";
        break;
    
    default:
        $nomeGiorno = "Giorno non valido";
        break;
}

echo "Il giorno corrispondente al numero $giorno è $nomeGiorno.";

// Dichiarazione di un array
$numeri = array(); // Dichiarazione di un array in PHP
$nomi = array(); // Dichiarazione di un array in PHP

// Inizializzazione di un array
$numeri = array(1, 2, 3, 4, 5); // Creazione di un array inizializzato con valori
$nomi = array("Alice", "Bob", "Charlie"); // Creazione di un array inizializzato con valori

// Accesso agli elementi di un array
$primoNumero = $numeri[0]; // Accesso al primo elemento dell'array
$secondoNome = $nomi[1]; // Accesso al secondo elemento dell'array

// Modifica degli elementi di un array
$numeri[2] = 42; // Modifica il terzo elemento dell'array
$nomi[0] = "Eve"; // Modifica il primo elemento dell'array

// Ottenere la lunghezza di un array
$lunghezzaNumeri = count($numeri); // Restituisce la lunghezza dell'array numeri
$lunghezzaNomi = count($nomi); // Restituisce la lunghezza dell'array nomi

// Iterazione attraverso un array
foreach ($numeri as $numero) {
    echo $numero; // Stampa ciascun elemento dell'array numeri
}

foreach ($nomi as $nome) {
    echo $nome; // Stampa ciascun elemento dell'array nomi
}

// Array multidimensionali
$matrice = array(
    array(1, 0, 0),
    array(0, 2, 0),
    array(0, 0, 3)
);
$valore = $matrice[2][2]; // Accesso all'elemento (2, 2) della matrice

// Lista (equivalente a un array dinamico)
$nomi = array();
array_push($nomi, "Alice");
array_push($nomi, "Bob");

// Classe e oggetto
class Persona {
    public $Nome;
    public $Età;
}

$persona1 = new Persona();
$persona1->Nome = "Maria";
$persona1->Età = 30;
?>
